porch/controllers/packagevariants/pkg/controllers/packagevariant/packagevariant_controller.go

```diff
@@ -782,12 +785,73 @@ func (r *PackageVariantReconciler) calculateDraftResources(ctx context.Context,
 		return nil, false, err
 	}
 
-	afterHash, err := hashFromPackageRevisionResources(&prr)
+	if len(prr.Spec.Resources) != len(origResources) {
+		// files were added or deleted
+		klog.Infoln(fmt.Sprintf("PackageVariant %q, PackageRevision %q, resources changed: %d original files, %d new files", pv.Name, prr.Name, len(origResources), len(prr.Spec.Resources)))
+		return &prr, true, nil
+	}
+
+	for k, v := range origResources {
+		newValue, ok := prr.Spec.Resources[k]
+		if !ok {
+			// a file was deleted
+			klog.Infoln(fmt.Sprintf("PackageVariant %q, PackageRevision %q, resources changed: %q in original files, not in new files", pv.Name, prr.Name, k))
+			return &prr, true, nil
+		}
+
+		if newValue != v {
+			// HACK ALERT - TODO(jbelamaric): Fix this
+			// Currently nephio controllers and package variant controller are rendering Kptfiles slightly differently in YAML
+			// not sure why, need to investigate more. It may be due to different versions of kyaml. So, here, just for Kptfiles,
+			// we will parse and compare semantically.
```

Comment:
I asked chatgpt to help :) and it generated the following.


```Go

package main

import (
	"fmt"
	"reflect"

	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
	"k8s.io/apimachinery/pkg/util/yaml"
)

func main() {
	// Example YAML strings
	yamlString1 := `
apiVersion: v1
kind: Pod
metadata:
  name: pod-1
spec:
  containers:
  - name: container-1
    image: nginx:latest
  - name: container-2
    image: redis:latest
`
	yamlString2 := `
apiVersion: v1
kind: Pod
metadata:
  name: pod-2
spec:
  containers:
  - name: container-1
    image: nginx:latest
  - name: container-2
    image: redis:latest
`

	// Convert YAML strings to unstructured objects
	obj1, err := yaml.ToUnstructured([]byte(yamlString1))
	if err != nil {
		panic(err)
	}

	obj2, err := yaml.ToUnstructured([]byte(yamlString2))
	if err != nil {
		panic(err)
	}

	// Compare unstructured objects
	equal := reflect.DeepEqual(obj1, obj2)
	fmt.Println("Objects are equal:", equal)
}
```

Reply:
:)